src/common/signalk/SKNMEAVisitor.cpp

```diff
@@ -68,12 +73,98 @@ void SKNMEAVisitor::visitSKNavigationAttitude(const SKUpdate &u, const SKPath &p
   sb.setField(8, "ROLL");
 
   _sentences.add(sb.toNMEA() + "\r\n");
+  DEBUG( sb.toNMEA().c_str() );
 };
 
 void SKNMEAVisitor::visitSKNavigationHeadingMagnetic(const SKUpdate &u, const SKPath &p, const SKValue &v) {
-  NMEASentenceBuilder sb2("II", "HDM", 2);
-  sb2.setField(1, SKRadToDeg(v.getNumberValue()), 1);
-  sb2.setField(2, "M");
+  NMEASentenceBuilder sb("II", "HDM", 2);
+  sb.setField(1, SKRadToDeg(v.getNumberValue()), 1);
+  sb.setField(2, "M");
 
-  _sentences.add(sb2.toNMEA() + "\r\n");
+  _sentences.add(sb.toNMEA() + "\r\n");
+  DEBUG( sb.toNMEA().c_str() );
 };
+
+// ***********************  Wind Speed and Angle  ************************
+//  Talker ID: WI Weather Instruments
+//             VW Velocity Sensor, Mechanical?
+//  also seen: IIVWR
+//
+//  NMEA0183  MWV Wind Speed and Angle  --> Deprecated
+//              1  2  3  4
+//              |  |  |  |
+//      $--MWV,x.x,a,x.x,a*hh
+//  1) Wind Angle, 0 to 360 degrees
+//  2) Reference, R = Relative, T = True
+//  3) Wind Speed
+//  4) Wind Speed Units, K/M/N
+//  5) Status, A = Data Valid
+//
+//      VWR Relative Wind Speed and Angle  --> implemented
+//              1  2  3  4  5  6  7  8
+//              |  |  |  |  |  |  |  |
+//      $--VWR,x.x,a,x.x,N,x.x,M,x.x,K*hh
+//  1) Wind direction magnitude in degrees
+//  2) Wind direction Left/Right of bow
+//  3) Speed
+//  4) N = Knots
+//  5) Speed
+//  6) M = Meters Per Second
+//  7) Speed
+//  8) K = Kilometers Per Hour
+// ************************************************************************
+void SKNMEAVisitor::visitSKEnvironmentWindAngleApparent(const SKUpdate &u, const SKPath &p, const SKValue &v) {
+
+  float windAngle;
+  float windSpeed;
+  windAngle = SKRadToDeg(v.getNumberValue());
```

Comment:
I think we should do the math in radians here. Only convert at the end when we print.

Reply:
Yes, agreed.
I had it in the code to do the maths only once, but now, I cancelled the deprecated NMEA0183 sentence.
(To be honest, I do not know what is heavier for CPU doing calculations twice or invent a new variable and do it once)